lib/Mojolicious/Guides/Tutorial.pod

```diff
@@ -0,0 +1,938 @@
+  get '/hello';
+
+  app->start;
+  __DATA__
+
+  @@ index.html.ep
+  <%= link_to Hello  => 'hello' %>.
+  <%= link_to Reload => 'index' %>.
+
+  @@ hello.html.ep
+  Hello World!
+
+Nameless routes get an automatically generated one assigned that is simply equal to the route itself without non-word
+characters.
+
+=head2 Layouts
+
+Templates can have layouts too, you just select one with the helper L<Mojolicious::Plugin::DefaultHelpers/"layout"> and
+place the result of the current template with the helper L<Mojolicious::Plugin::DefaultHelpers/"content">.
+
+  use Mojolicious::Lite;
+
+  get '/with_layout';
+
+  app->start;
+  __DATA__
+
+  @@ with_layout.html.ep
+  % title 'Green';
+  % layout 'green';
+  Hello World!
+
+  @@ layouts/green.html.ep
+  <!DOCTYPE html>
+  <html>
+    <head><title><%= title %></title></head>
+    <body><%= content %></body>
+  </html>
+
+The stash or helpers like L<Mojolicious::Plugin::DefaultHelpers/"title"> can be used to pass additional data to the
+layout.
+
+=head2 Blocks
+
+Template blocks can be used like normal Perl functions and are always delimited by the C<begin> and C<end> keywords,
+they are the foundation for many helpers.
+
+  use Mojolicious::Lite;
+
+  get '/with_block' => 'block';
+
+  app->start;
+  __DATA__
+
+  @@ block.html.ep
+  % my $link = begin
+    % my ($url, $name) = @_;
+    Try <%= link_to $url => begin %><%= $name %><% end %>.
+  % end
+  <!DOCTYPE html>
+  <html>
+    <head><title>Sebastians frameworks</title></head>
+    <body>
+      %= $link->('http://mojolicious.org', 'Mojolicious')
+      %= $link->('http://mojojs.org', 'mojo.js')
+    </body>
+  </html>
+
+=head2 Helpers
+
+Helpers are little functions you can create with the keyword L<Mojolicious::Lite/"helper"> and reuse throughout your
+whole application, from actions to templates.
+
+  use Mojolicious::Lite -signatures;
+
+  # A helper to identify visitors
+  helper whois => sub ($c) {
+    my $agent = $c->req->headers->user_agent || 'Anonymous';
+    my $ip    = $c->tx->remote_address;
+    return "$agent ($ip)";
+  };
+
+  # Use helper in action and template
+  get '/secret' => sub ($c) {
+    my $user = $c->whois;
+    $c->app->log->debug("Request from $user");
+  };
+
+  app->start;
+  __DATA__
+
+  @@ secret.html.ep
+  We know who you are <%= whois %>.
+
+A list of all built-in ones can be found in L<Mojolicious::Plugin::DefaultHelpers> and
+L<Mojolicious::Plugin::TagHelpers>.
+
+=head2 Plugins
+
+Plugins are application extensions that help with code sharing and organization. You can load a plugin with the keyword
+L<Mojolicious::Lite/"plugin">, which can omit the C<Mojolicious::Plugin::> part of the name, and optionally provide
+configuration for the plugin.
+
+  use Mojolicious::Lite;
+
+  plugin Config => {file => '/etc/myapp.conf', default => {foo => 'bar'}};
+
+  # Return configured foo value, or default if no configuration file
+  get '/foo' => sub ($c) {
+    my $foo = $c->app->config('foo');
```

Comment:
What's the advice for the best interface to use?  `app->config('foo')` or `app->config->{foo}`?  I get they are both options via `stash`, but I assume the latter is more performant and recommend to use when practical.  What is the value to using the form `app->config('foo')`?
Can there be a "Pointer" capability available here, too?  `app->config('/foo/bar')`